wallet/src/Pos/Util/Mnemonic.hs

```diff
@@ -75,17 +75,16 @@ data Mnemonic (mw :: Nat) = Mnemonic
 -- ERRORS
 --
 
-data MnemonicException = UnexpectedMnemonicErr MnemonicErr
+data MnemonicException csz = UnexpectedMnemonicErr (MnemonicErr csz)
     deriving (Show, Typeable)
 
 
-data MnemonicErr
-    = MnemonicErrInvalidEntropyLength Int
-    | MnemonicErrFailedToCreate
+data MnemonicErr csz
+    = MnemonicErrInvalidWord DictionaryError
+    | MnemonicErrInvalidEntropy (EntropyError csz)
```

Comment:
As we discussed, this can do:

```hs
data MnemonicError csz
    = ErrMnemonicWords MnemonicWordsError
    | ErrEntropy (EntropyError csz)
    | ErrDictionnary DictionaryError
    | ErrForbidden
    deriving (Show)
```